src/main/java/com/simondata/pouroversql/clients/ExtractorEngine.java

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019-present, Simon Data, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.simondata.pouroversql.clients;
+
+import org.apache.commons.lang3.EnumUtils;
+
+/**
+ * <h1>ExtractorEngine</h1>
+ * The type of Extractor to use.
+ */
+public enum ExtractorEngine {
+    SFTP,
+    SQL;
+
+    private static String normalizeName(String name) {
+        String result;
+        String cleaned = name.toUpperCase().replaceAll(" ", "_");
+        switch (cleaned) {
+            case "SFTP":
+                result = SFTP.name();
+                break;
+            case "SQL":
+                result = SQL.name();
+                break;
+            default:
+                result = name.toUpperCase();
+                break;
+        }
+        return result;
+    }
+
+    /**
+     * Safer version of converting a string to enum.
+     * Resiliant against differences in common names (MySQL vs MariaDB)
```

Comment:
The only choices that I see are "SQL" and "SFTP"?

Reply:
yeah, just the two; was leaving the convention the same in case more were ever added, but updated the comment to not reference database names